packages/mui-base/src/useSwitch/useSwitch.types.ts

```diff
@@ -49,3 +49,27 @@ interface UseSwitchInputSlotOwnProps {
 
 export type UseSwitchInputSlotProps<TOther = {}> = Omit<TOther, keyof UseSwitchInputSlotOwnProps> &
   UseSwitchInputSlotOwnProps;
+
+export interface UseSwitchReturnValue {
+  /**
+   * If `true`, the component will be checked.
+   */
+  checked: boolean;
+  /**
+   * If `true`, the component will be disabled.
+   */
+  disabled: boolean;
+  /**
+   * If `true`, it indicates that the component is being focused using keyboard.
+   * @default false
+   */
+  focusVisible: boolean;
+  /**
+   * Function to get the props to place on an HTML `input` to make the switch accessible.
+   */
+  getInputProps: (otherProps?: React.HTMLAttributes<HTMLInputElement>) => UseSwitchInputSlotProps;
```

Comment:
In order to not repeat the types, can you define a type for `getInputProps` in `useSwitch.types.ts` and type in both the return definition here and in the `getInputProps` implementation?

Reply:
But does it get changed in the type column in the API docs here? If yes, we should not extract.

Reply:
I reused the `getInputProps` in commit https://github.com/mui/material-ui/pull/36050/commits/d2d2545333f9d7f9a77b71ce3ecfd64cf3ccec68 also ensuring the docs shows the type.